batch-nft-reveal/app/hooks/useOwnedTokens.ts

```diff
@@ -0,0 +1,29 @@
+import { useContractCall } from '../hooks/useContractCall'
+import { useContractCalls } from '@usedapp/core'
+import json from '../artifacts/contracts/NFTCollection.sol/NFTCollection.json'
+import { BigNumber, utils } from 'ethers'
+
+export function useOwnedTokens(
+  contractAddr: string,
+  account: string
+): Array<Array<BigNumber>> {
+  const ownerBalanceCall: BigNumber = useContractCall(
+    'balanceOf',
+    [account],
+    contractAddr
+  )
+  const ownedTokensCalls = []
+  const { abi } = json
+
+  const ownerBalance = ownerBalanceCall && ownerBalanceCall.toNumber()
+  for (let i = ownerBalance - 1; i >= 0; i--) {
```

Comment:
It's more straightforward to iterate starting from the beginning and return the result ordered by index and then do the sorting in the component. The hook is also more reusable that way.